### PR TITLE
Remove dead code

### PR DESCRIPTION
**Pull request by rubenwardy**
There's a lot of unused code in Minetest - whether that be disabled using `#ifdef 0`, not called, or not defined at all.

This PR removes dead code whilst leaving unused code which may be useful in the future. Keeping dead code around is bad for code quality and can be a maintenance burden.

Let's not continue to hoard code from 2012

## To do

Ready for review

## How to test

See if build completes

### CODE REVIEW

File: src/client/tile.cpp [hunk 1]
```diff
@@ -450,7 +449,7 @@ TextureSource::TextureSource()
 	// for these settings to take effect
 	m_setting_trilinear_filter = g_settings->getBool("trilinear_filter");
 	m_setting_bilinear_filter = g_settings->getBool("bilinear_filter");
-	m_setting_anisotropic_filter = g_settings->getBool("anisotropic_filter");
+	g_settings->getBool("anisotropic_filter");
```
Comment (sfan5) [inline src/client/tile.cpp:452]:
```suggestion
```

File: src/client/client.cpp [hunk 2]
```diff
@@ -216,12 +216,6 @@ void Client::loadMods()
 		m_script->on_minimap_ready(m_minimap);
 }
 
-bool Client::checkBuiltinIntegrity()
-{
-	// TODO
-	return true;
```
Comment (sfan5) [inline src/client/client.cpp:222]: should also remove the commented-out code that calls this

Comment (rubenwardy) [reply src/client/client.cpp:222]: I was tied over whether to delete and it and left it due to the TODO. Deleted now

**Comment by kilbith** [comment]
Could include https://github.com/minetest/minetest/pull/10651

**Comment by rubenwardy** [comment]
Bet this breaks things on some platform

**Comment by rubenwardy** [comment]
> Could include #10651

Technically not dead code. I tried doing it now but my IDE doesn't make it trivial so I think it should be left to another PR

File: src/emerge.cpp [hunk 3]
```diff
@@ -396,13 +396,6 @@ int EmergeManager::getGroundLevelAtPoint(v2s16 p)
...
 bool EmergeManager::isBlockUnderground(v3s16 blockpos)
 {
-#if 0
-	v2s16 p = v2s16((blockpos.X * MAP_BLOCKSIZE) + MAP_BLOCKSIZE / 2,
-					(blockpos.Y * MAP_BLOCKSIZE) + MAP_BLOCKSIZE / 2);
-	int ground_level = getGroundLevelAtPoint(p);
-	return blockpos.Y * (MAP_BLOCKSIZE + 1) <= min(water_level, ground_level);
-#endif
-
 	// Use a simple heuristic; the above method is wildly inaccurate anyway.
```
Comment (randomMesh) [inline src/emerge.cpp:399]: I'd remove the part after the semicolon in the comment since there is no "above method" anymore.

Comment (rubenwardy) [reply src/emerge.cpp:399]:
```suggestion
	// Use a simple heuristic
```

File: src/emerge.cpp [hunk 4]
```diff
@@ -396,7 +396,7 @@ int EmergeManager::getGroundLevelAtPoint(v2s16 p)
 // TODO(hmmmm): Move this to ServerMap
 bool EmergeManager::isBlockUnderground(v3s16 blockpos)
```
Comment (randomMesh) [inline src/emerge.cpp:397]: The parameter `blockpos` can be a const reference. May be offtopic for this pull request.

**Review by sfan5 — APPROVED** [review]
LGTM

**Pull request merged**